tensorflow/core/lib/io/buffered_random_access_file_test.cc

```diff
@@ -0,0 +1,140 @@
+#include "tensorflow/core/lib/io/buffered_random_access_file.h"
+
+#include "tensorflow/core/lib/core/status_test_util.h"
+#include "tensorflow/core/lib/random/random.h"
+#include "tensorflow/core/platform/env.h"
+#include "tensorflow/core/platform/test.h"
+#include "tensorflow/core/platform/test_benchmark.h"
+
+namespace tensorflow {
+namespace io {
+
+class TestBufferedRandomAccessFile : public ::testing::TestWithParam<size_t> {
+ protected:
+  void SetUp() override {
+    buf_size_ = GetParam();
+    env_ = Env::Default();
+    fname_ = testing::TmpDir() + "/buffered_random_access_file_test";
+    StringPiece element = elements_.substr(0, 10);  // "0123456789"
+
+    {
+      // create a file containing as many as `file_len` repeating digits.
+      env_->DeleteFile(fname_).IgnoreError();
+      std::unique_ptr<WritableFile> writable;
+      TF_ASSERT_OK(env_->NewAppendableFile(fname_, &writable));
+      for (int i = 0; i < file_len_ / element.length(); i++) {
+        TF_ASSERT_OK(writable->Append(element));
+      }
+      TF_ASSERT_OK(writable->Close());
+    }
+  }
+  void TearDown() override {}
+
+ protected:
+  Env* env_;
+  const size_t file_len_ = 1000;  // make sure file_len_ % 10 == 0
+  const StringPiece elements_ =
+      "01234567890123456789012345678901234567890123456789";  // len = 50
+  size_t buf_size_;
+  string fname_;
+};
+
+TEST_P(TestBufferedRandomAccessFile, SequencialRead) {
+  std::unique_ptr<RandomAccessFile> raf;
+  TF_ASSERT_OK(env_->NewRandomAccessFile(fname_, &raf));
+  char scratch[1024];
+
+  BufferedRandomAccessFile braf(std::move(raf), buf_size_);
+  StringPiece result;
+  uint64 offset = 0;
+  while (offset < file_len_) {
+    // may read more than a buffer size.
+    size_t n = 1 + random::New64() % (buf_size_ * 2);
+    // read exactly to file end.
+    n = std::min<size_t>(n, file_len_ - offset);
+    // std::cout << "read >>>  n: " << n << ", offset: " << offset << std::endl;
```

Comment:
We should remove the debug prints, even if commented

Reply:
removed.